hazelcast/src/main/java/com/hazelcast/client/impl/spi/ProxyManager.java

```diff
@@ -330,13 +388,58 @@ private void initialize(ClientProxy clientProxy) throws Exception {
     }
 
     public Collection<? extends DistributedObject> getDistributedObjects() {
+        getAndProcessDistributedObjects();
+        return getLocalDistributedObjects();
+    }
+
+    private void getAndProcessDistributedObjects() {
```

Comment:
Should we add a `throws RuntimeException` here? I am not sure because every method does not have it.